### ISSUE TITLE
[3.x] C# callback methods with parameters crash randomly

### ISSUE THREAD

**Issue by zaevi** [godotengine/godot#56385]
### Godot version

3.5.beta.mono (7fcfde9)

### System information

Win10

### Issue description

My project crashed randomly in 3.5.beta. It happens when engine passes parameters to managed world.

```
E 0:00:01.113   GDMonoUtils::debug_send_unhandled_exception_error: System.StackOverflowException: The requested operation caused a stack overflow.
  <C++ 错误>      Unhandled exception
  <C++ 源文件>     modules\mono\mono_gd\gd_mono_utils.cpp:422 @ GDMonoUtils::debug_send_unhandled_exception_error()
```

May related to #53942 since it's commits about boxing/unboxing.

### Steps to reproduce

It will randomly crash at starting:
```csharp
public class Main : Godot.Node2D
{
    public override void _Ready() { }

    public override void _Process(float delta) { } // if remove this line, the project will run fine
}
```

More obvious case. When pressing any key, it will randomly crash:
```csharp
public class Main : Godot.Node2D
{
    public override void _UnhandledKeyInput(Godot.InputEventKey ev) { }
}
```

**Comment by raulsntos** [comment]
#53942 is indeed the cause, I looked a bit more into it and found that I forgot to initialize the `params_buffer_size` member :disappointed: so I went ahead and opened #56391 to fix it.

**Comment by akien-mga** [comment]
Fixed by #56391.
